### PR TITLE
42 while waiting for data to come show loading

### PR DESCRIPTION
**Pull request by ElhamMoharrami**

### CODE REVIEW

File: src/pages/ArtistDetail/ArtistDetail.js [hunk 1]
```diff
@@ -8,11 +8,11 @@ import { useParams } from "react-router-dom";
...
 
-import classes from "./ArtistDetailPage.module.css";
+import classes from "./ArtistDetail.module.css";
 import blankProfile from "../../assets/images/blankProfilePicture.png";
 import { artistActions } from "../../store/data-slice";
 
-const ArtistDetailPage = () => {
+const ArtistDetail = () => {
   const card = (item) => {
     return <MovieCard movie={item} key={item.id} />;
```
Comment (ghamilouie) [inline src/pages/ArtistDetail/ArtistDetail.js:17]: Isn't it better move `key={item.id` in showList

File: src/pages/Artists.js [hunk 2]
```diff
@@ -8,13 +8,14 @@ import { artistActions } from "../store/data-slice";
 
 const Artists = () => {
   const data = useSelector((state) => state.crews.data);
+  const isLoading=useSelector((state)=>state.crews.isLoading)
 
   const card = (item) => {
     return <ArtistCard artist={item} key={item.id} />;
```
Comment (ghamilouie) [inline src/pages/Artists.js:14]: same for `key={item.id`

File: src/pages/MovieDetail/MovieDetail.js [hunk 3]
```diff
@@ -98,7 +97,7 @@ const MovieDetailPage = () => {
             itemClass="carousel-item-padding-40-px"
           >
             {artists.map((item) => (
-              <ArtistCard artist={item} />
+              <ArtistCard artist={item} key={item.id} />
```
Comment (ghamilouie) [inline src/pages/MovieDetail/MovieDetail.js:100]: same for `key={item.id`

File: src/pages/Movies.js [hunk 4]
```diff
@@ -9,12 +9,13 @@ import { movieActions } from "../store/data-slice";
 
 const Movies = () => {
   const data = useSelector((state) => state.movies.data);
+  const isLoading=useSelector((state)=>state.movies.isLoading)
   const card = (item) => {
     return <MovieCard movie={item} key={item.id} />;
```
Comment (ghamilouie) [inline src/pages/Movies.js:14]: same for `key={item.id`

File: src/pages/MovieDetail/MovieDetail.js [hunk 5]
```diff
@@ -86,7 +85,7 @@ const MovieDetailPage = () => {
             draggable={false}
             showDots={true}
             responsive={responsive}
-            ssr={true} // means to render carousel on server-side.
+            ssr={true}
```
Comment (ghamilouie) [inline src/pages/MovieDetail/MovieDetail.js:88]: It is better to create new component ex. `carouselShowList`, you might use it in other pages

**Review by ghamilouie — CHANGES_REQUESTED** [review]

File: src/pages/Movies.js (hunk 4 above)
Comment (ElhamMoharrami) [reply src/pages/Movies.js:14]: I don't quite understand what you mean. I can't find any suitable position fir key in ShowList component.

**Pull request merged**